standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -6052,450 +1096,53 @@ protected synchronized static DataSource setupJdbcConnectionPool(Configuration c
-          LOG.error(msg);
-          throw new RuntimeException(msg);
-        }
-        try {
-          LOG.info("Going to load JDBC driver {}", driverName);
-          driver = (Driver) Class.forName(driverName).newInstance();
-        } catch (InstantiationException e) {
-          throw new RuntimeException("Unable to instantiate driver " + driverName + ", " +
-              e.getMessage(), e);
-        } catch (IllegalAccessException e) {
-          throw new RuntimeException(
-              "Unable to access driver " + driverName + ", " + e.getMessage(),
-              e);
-        } catch (ClassNotFoundException e) {
-          throw new RuntimeException("Unable to find driver " + driverName + ", " + e.getMessage(),
-              e);
-        }
-        connString = MetastoreConf.getVar(conf, ConfVars.CONNECT_URL_KEY);
-      }
-
-      try {
-        LOG.info("Connecting to transaction db with connection string {}", connString);
-        Properties connectionProps = new Properties();
-        connectionProps.setProperty("user", username);
-        connectionProps.setProperty("password", password);
-        Connection conn = driver.connect(connString, connectionProps);
-        String prepareStmt = dbProduct != null ? dbProduct.getPrepareTxnStmt() : null;
-        if (prepareStmt != null) {
-          try (Statement stmt = conn.createStatement()) {
-            stmt.execute(prepareStmt);
-          }
-        }
-        conn.setAutoCommit(false);
-        return conn;
-      } catch (SQLException e) {
-        throw new RuntimeException("Unable to connect to transaction manager using " + connString
-            + ", " + e.getMessage(), e);
-      }
-    }
-
-    @Override
-    public PrintWriter getLogWriter() throws SQLException {
-      throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public void setLogWriter(PrintWriter out) throws SQLException {
-      throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public void setLoginTimeout(int seconds) throws SQLException {
-      throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public int getLoginTimeout() throws SQLException {
-      throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public java.util.logging.Logger getParentLogger() throws SQLFeatureNotSupportedException {
-      throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public <T> T unwrap(Class<T> iface) throws SQLException {
-      throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public boolean isWrapperFor(Class<?> iface) throws SQLException {
-      throw new UnsupportedOperationException();
-    }
+    jdbcResource.getJdbcTemplate().getJdbcTemplate().execute((Statement stmt) -> {
+        stmt.execute(sqlStmt);
+        return null;
+    });
+    LOG.debug("TXN lock locked by '{}' in mode {}", TxnHandler.hostname, shared);
   }
 
   @Override
   @RetrySemantics.SafeToRetry
```

Comment:
should we define retry semantic here, as it's also present in interface?

Reply:
removed